File: LeetCode/Binary_tree_level_order_traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int getHeight(TreeNode *root){
        if(!root) return 0;
        else return max(getHeight(root->left), getHeight(root->right))+1;
    }
    void levelOrderUtil(TreeNode *root, vector<vector<int>> &result, int level){
        if(!root) return;
        else{
            result[level].push_back(root->val);
            levelOrderUtil(root->left,result,level+1);
            levelOrderUtil(root->right,result,level+1);
        }
    }
    vector<vector<int>> levelOrder(TreeNode* root) {
        int height = getHeight(root);
        vector<vector<int>> result(height);
        levelOrderUtil(root,result,0);
        return result;
    }
};

File: LeetCode/binary_tree_level_order_traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        ArrayList<List<Integer>> result = new ArrayList<List<Integer>>();
        levelOrderUtil(root,0, result);
        return result;
    }
    
    public void levelOrderUtil(TreeNode root, int currentLevel, List<List<Integer>> result){
        if(root == null) return;
        else{
            if(currentLevel == result.size()){
                 result.add(new ArrayList<Integer>());
            }
            levelOrderUtil(root.left, currentLevel+1, result);
            (result.get(currentLevel)).add(root.val);
            levelOrderUtil(root.right, currentLevel+1, result);
        }     
    }
}